### PR TITLE
feat:[CDS-80838]: add support for search in overrides

### PR DESCRIPTION
**Pull request by sourabh-awashti**
## Describe your changes
Changes are regarding the seach support on identifiers for overrides v2.
Tests Done:
1. Search with env, svc, infra ref for overrides for all types of overrides
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55139)

### CODE REVIEW

**Comment by sourabh-awashti** [comment]
trigger uts

**Review by abhinavmittal-harness — APPROVED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/55139)* status: 4 file(s) to review. 1 of 2 approvals obtained.

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverrideCriteriaHelper.java [hunk 1]
```diff
@@ -31,13 +33,18 @@ public class ServiceOverrideCriteriaHelper {
...
+      ServiceOverridesType type, String searchTerm, OverrideFilterPropertiesDTO filterProperties) {
     Criteria criteria = new Criteria();
     criteria.and(ACCOUNT_ID).is(accountId);
     criteria.and(ORG_ID).is(orgIdentifier);
     criteria.and(PROJECT_ID).is(projectIdentifier);
     criteria.and(NGServiceOverridesEntity.NGServiceOverridesEntityKeys.spec).exists(true).ne(null);
 
+    if (isNotEmpty(searchTerm)) {
+      criteria.and(NGServiceOverridesEntityKeys.identifier)
+          .regex(searchTerm, NGResourceFilterConstants.CASE_INSENSITIVE_MONGO_OPTIONS);
```
Comment (lovish1999) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverrideCriteriaHelper.java:45]: but identifier also includes "_". Won't that cause any issues if given in filter ?

Comment (sourabh-awashti) [reply 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverrideCriteriaHelper.java:45]: No @lovish1999 , we are checking with regex and not with the exact string match

**Comment by sourabh-awashti** [comment]
trigger codeformat

**Review by lovish1999 — APPROVED** [review]
LGTM

**Comment by sourabh-awashti** [comment]
trigger checkstylecodeformat

**Comment by sourabh-awashti** [comment]
trigger uts

**Pull request merged**